src/v/storage/disk_log_impl.cc

```diff
@@ -1747,7 +1747,7 @@ void disk_log_impl::bg_checkpoint_offset_translator() {
 ss::future<> disk_log_impl::force_roll(ss::io_priority_class iopc) {
     auto roll_lock_holder = co_await _segments_rolling_lock.get_units();
     auto t = term();
-    auto next_offset = offsets().dirty_offset + model::offset(1);
+    auto next_offset = model::next_offset(offsets().dirty_offset);
```

Comment:
how is this related to the race?

Reply:
Unrelated to the race, related to the test case I wrote in which I call `force_roll()` on a log that has nothing produced to it, `offsets.dirty_offset() == -9223372036854775808`, and the offset value `-9223372036854775808 + 1` is going to trigger an assert in [`disk_log_impl::new_segment()`](https://github.com/redpanda-data/redpanda/blob/60eadf0cb406ca16419537b96b819c8b6440e0e5/src/v/storage/disk_log_impl.cc#L1613-L1614).

If, instead, we use `model::next_offset(-9223372036854775808) == 0`, we are okay :+1:

Reply:
ahh, got it. i forgot that force_roll is test-only code. yolo!